### PR TITLE
gitoxide: Install shell completions.

### PR DESCRIPTION
**Pull request by sbatial**
## Description of changes

`gix` has a subcommand to generate shell completions.
These are now installed in the prefixup phase.
I took the logic from how `rg` installs their completions as that's a rust project using clap as well.

The second binary `ein` does not ~~(yet, I've opened a PR about this in the upstream repo as well: https://github.com/Byron/gitoxide/pull/1296)~~ have the ability to generate completions, in the version on nixpkgs but will use the same commands (except for the name of course).
That should be the case with the next release I suppose.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by ambroisie — APPROVED** [review]

File: pkgs/applications/version-management/gitoxide/default.nix [hunk 1]
```diff
@@ -26,10 +31,17 @@ rustPlatform.buildRustPackage rec {
 
   nativeBuildInputs = [ cmake pkg-config ];
 
-  buildInputs = [ curl ] ++ (if stdenv.isDarwin
+  buildInputs = [ curl installShellFiles ] ++ (if stdenv.isDarwin
```
Comment (natsukium) [inline pkgs/applications/version-management/gitoxide/default.nix:34]: Could you please move `installShellFiles` to `nativeBuildInputs`? 
It is not a run-time dependency but a build-time dependency.

Comment (sbatial) [reply pkgs/applications/version-management/gitoxide/default.nix]: Done in https://github.com/NixOS/nixpkgs/pull/289330/commits/b9c5c429f6d7169e48cf287cefd1afec57f38b18

**Comment by sbatial** [comment]
I just realised, that my commit b9c5c429f6d7169e48cf287cefd1afec57f38b18 contained more than just the move of the buildInputs.
I removed it again because the completions for `ein` are not available for gitoxide v0.33.0.

But [gitoxide v0.34.0](https://github.com/Byron/gitoxide/releases/tag/v0.34.0) contains completions for `ein` as well, so I'll open another PR which would upgrade the version and add the completions for `ein`.

**Review by natsukium — APPROVED** [review]
LGTM, thanks for your contribution!

**Pull request merged**